sections/main-product.liquid

```diff
@@ -693,6 +693,100 @@
                   </span>
                 </p>
               {%- endif -%}
+
+              {%- when 'complementary' -%}
+                <div class="complementary-products quick-add-hidden no-js-hidden{% if block.settings.make_collapsible_row %} product__accordion accordion{% endif %}" {{ block.shopify_attributes }}>
+                  <product-recommendations data-url="{{ routes.product_recommendations_url }}?section_id={{ section.id }}&product_id={{ product.id }}&limit={{ block.settings.product_list_limit }}&intent=complementary">
+                    <div class="complementary-products__container">
+                      <div class="skeleton product-card-wrapper contains-product-card">
+                        <div class="skeleton__heading"><span></span></div>
+                        {%- for i in (1..block.settings.products_per_page) -%}
+                          <div class="skeleton__container card card--horizontal card--{{ settings.card_style }}{% if settings.card_style == 'card' %} color-background-1 gradient{% endif %}">
+                            <div class="skeleton__image skeleton__image--{{ block.settings.image_ratio }} card__inner"><span></span></div>
+                            <div class="skeleton__content card__content">
+                              <div><span></span></div>
+                              <div><span></span></div>
+                            </div>
+                          </div>
+                        {%- endfor -%}
+                      </div>
+                      {%- if recommendations.performed and recommendations.products_count > 0 -%}
+                        {%- if block.settings.make_collapsible_row -%}
+                          <details id="Details-{{ block.id }}-{{ section.id }}" open>
```

Comment:
@metamoni when I check it with voice over it does work and announces it properly for me: [video](https://screenshot.click/07-02-9byc6-ynev1.mp4)

I wonder if there is an issue triggered due to the previous collapsible rows no containing anything and messing with the component code to update the attributes properly. But as you can see in the video when you get to the complementary product as a collapsible row, the state is conveyed properly.

Reply:
Interesting, it's still not working for me. But it's not a component you wrote from scratch, so it's not a blocking comment
https://user-images.githubusercontent.com/22390758/188917148-388e5165-191c-4ad4-90a8-55855f7facca.mov